pkg/api/latest/latest.go

```diff
@@ -18,105 +18,87 @@ package latest
 
 import (
 	"fmt"
-	"strings"
 
-	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/registered"
-	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/runtime"
-	"k8s.io/kubernetes/pkg/util/sets"
 )
 
-// Version is the string that represents the current external default version.
-var Version string
-
-// OldestVersion is the string that represents the oldest server version supported,
-// for client code that wants to hardcode the lowest common denominator.
-var OldestVersion string
-
-// Versions is the list of versions that are recognized in code. The order provided
-// may be assumed to be least feature rich to most feature rich, and clients may
-// choose to prefer the latter items in the list over the former items when presented
-// with a set of versions to choose.
-var Versions []string
-
-// Codec is the default codec for serializing output that should use
-// the latest supported version.  Use this Codec when writing to
-// disk, a data store that is not dynamically versioned, or in tests.
-// This codec can decode any object that Kubernetes is aware of.
-var Codec runtime.Codec
-
-// accessor is the shared static metadata accessor for the API.
-var accessor = meta.NewAccessor()
-
-// SelfLinker can set or get the SelfLink field of all API types.
-// TODO: when versioning changes, make this part of each API definition.
-// TODO(lavalamp): Combine SelfLinker & ResourceVersioner interfaces, force all uses
-// to go through the InterfacesFor method below.
-var SelfLinker = runtime.SelfLinker(accessor)
-
-// RESTMapper provides the default mapping between REST paths and the objects declared in api.Scheme and all known
-// Kubernetes versions.
-var RESTMapper meta.RESTMapper
-
-// userResources is a group of resources mostly used by a kubectl user
-var userResources = []string{"rc", "svc", "pods", "pvc"}
-
-const importPrefix = "k8s.io/kubernetes/pkg/api"
-
-func init() {
-	// Use the first API version in the list of registered versions as the latest.
-	Version = registered.RegisteredVersions[0]
-	OldestVersion = registered.RegisteredVersions[len(registered.RegisteredVersions)-1]
-	Codec = runtime.CodecFor(api.Scheme, Version)
-	// Put the registered versions in Versions in reverse order.
-	versions := registered.RegisteredVersions
-	Versions = []string{}
-	for i := len(versions) - 1; i >= 0; i-- {
-		Versions = append(Versions, versions[i])
+var (
+	allGroups     = GroupMetaMap{}
+	Group         = allGroups.Group
+	RegisterGroup = allGroups.RegisterGroup
+	GroupOrDie    = allGroups.GroupOrDie
+)
+
+type GroupMetaMap map[string]*GroupMeta
+
+func (g GroupMetaMap) RegisterGroup(group string) (*GroupMeta, error) {
```

Comment:
Please add godoc-style comments to the public functions/methods/types in this file.